NEWS.md

```diff
@@ -1,251 +0,0 @@
-## DataFrames v0.6.11 Release Notes
```

Comment:
The old NEWS file will be in the Git history, so people can go looking for it if they want it. We can just add release notes in the actual GitHub releases going forward but no need to edit the past releases. Likely more effort than it's worth. Presumably people at this point are familiar with current and past tags and would be more interested in what changes next.